packages/browser/src/plugins/destinations/google-tag-manager.ts

```diff
@@ -79,16 +58,16 @@ const googleTagManager: DestinationFactory<GoogleTagManagerSettings> = ({
           .filter(Boolean)
           .join(' ')
 
-        window.gtag('event', eventName, {
-          ...baseEvent(ctx),
+        pushEvent({
+          ...ctx.event,
+          type: 'track',
+          event: eventName,
         })
       }
     },
 
-    track: (ctx) => {
-      window.gtag('event', ctx.event.event, {
-        ...baseEvent(ctx),
-      })
+    track: (ctx: Context) => {
+      pushEvent(ctx.event)
     },
```

Comment:
I think we could support `identify` here too?

```
    identify: (ctx: Context) => {
        dataLayer.push({ user_id: ctx.event.userId })
    },
```

https://developers.google.com/analytics/devguides/collection/ga4/user-id?client_type=gtm#step_1_update_your_data_layer